### PR TITLE
Added troubleshooting and note about required storage:delete:enabled flag when pruning imags

### PR DESCRIPTION
**Pull request by soltysh**
Encountered in https://bugzilla.redhat.com/show_bug.cgi?id=1339674 so I've added note and a troubleshooting element to address the problem.

@ahardin-rh ptal
@miminar fyi

### CODE REVIEW

File: admin_guide/pruning_resources.adoc [hunk 1]
```diff
@@ -166,6 +166,15 @@ xref:../architecture/additional_concepts/authorization.adoc#roles[cluster role]
 *system:image-pruner* or greater (for example, *cluster-admin*).
 ====
 
+[NOTE]
+====
+Pruning images above all, invokes removes data from the docker registry. For this operation
```
Comment (pweil-) [inline admin_guide/pruning_resources.adoc]: > Pruning images above all, invokes removes data 

Pruning images removes data ...

File: install_config/install/docker_registry.adoc [hunk 2]
```diff
@@ -1511,6 +1511,28 @@ configuration file] lacks mandatory entries in the
 xref:docker-registry-configuration-reference-middleware[middleware section].
 Add them, re-deploy the registry, and restart your builds.
 
+[[known-issue-prune-fails-due-to-delete-disabled]]
+==== Image Pruning Fails
+
+If you encounter following errors when pruning images:
```
Comment (pweil-) [inline install_config/install/docker_registry.adoc]: s/errors/error

File: install_config/install/docker_registry.adoc [hunk 3]
```diff
@@ -1511,6 +1511,28 @@ configuration file] lacks mandatory entries in the
...
+
+----
+BLOB sha256:49638d540b2b62f3b01c388e9d8134c55493b1fa659ed84e97cb59b87a6b8e6c error deleting blob
+----
+
+And your xref:viewing-logs[registry log] contains following information:
```
Comment (pweil-) [inline install_config/install/docker_registry.adoc]: contains the following

**Review by pweil- — CHANGES_REQUESTED** [review]

**Comment by soltysh** [comment]
@pweil- comments addressed

File: install_config/install/docker_registry.adoc [hunk 4]
```diff
@@ -1511,6 +1511,28 @@ configuration file] lacks mandatory entries in the
 xref:docker-registry-configuration-reference-middleware[middleware section].
 Add them, re-deploy the registry, and restart your builds.
 
+[[known-issue-prune-fails-due-to-delete-disabled]]
+==== Image Pruning Fails
+
+If you encounter following error when pruning images:
```
Comment (pweil-) [inline install_config/install/docker_registry.adoc]: the following

**Review by pweil- — APPROVED** [review]
I missed one :)  Other than that this LGTM

**Comment by soltysh** [comment]
One missed addressed as well :smiley:

File: install_config/install/docker_registry.adoc [hunk 5]
```diff
@@ -1511,6 +1511,28 @@ configuration file] lacks mandatory entries in the
...
+----
+
+And your xref:viewing-logs[registry log] contains the following information:
+
+----
+error deleting blob \"sha256:49638d540b2b62f3b01c388e9d8134c55493b1fa659ed84e97cb59b87a6b8e6c\": operation unsupported
+----
+
+It means that your xref:advanced-overriding-the-registry-configuration[custom
+configuration file] lacks mandatory entries in the
```
Comment (miminar) [inline install_config/install/docker_registry.adoc]: Please mark it as mandatory also in the [storage config section](https://docs.openshift.org/latest/install_config/install/docker_registry.html#docker-registry-configuration-reference-storage).

Comment (soltysh) [inline install_config/install/docker_registry.adoc]: I've put a comment next to that config option with info about it being mandatory for image pruning.

File: install_config/install/docker_registry.adoc [hunk 6]
```diff
@@ -1013,6 +1013,7 @@ storage:
     readonly:
       enabled: false
 ----
+<1> This entry is *mandatory* for image pruning to work properly.
```
Comment (miminar) [inline install_config/install/docker_registry.adoc]: Can you also explain, what it actually does (enables removal of data on registry's storage through API)?

Comment (soltysh) [inline install_config/install/docker_registry.adoc]: Do you think that the key naming `storage:delete:enabled` isn't sufficient enough?

**Comment by ahardin-rh** [comment]
@soltysh Looks good to me! Please double-check my labels for this. I added a to_followup so that our docs team can take care of a few style nits. Thanks!

**Comment by soltysh** [comment]
@ahardin-rh labels LGTM

File: install_config/install/docker_registry.adoc (hunk 6 above)
Comment (miminar) [inline install_config/install/docker_registry.adoc]: It wasn't immediately obvious to me. But that's my one and only argument. If you find it sufficient, I won't argue.

**Comment by miminar** [comment]
LGTM

**Comment by ahardin-rh** [comment]
Thanks!

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../admin_guide/pruning_resources.adoc#admin-guide-pruning-resources[Pruning Objects]
|Added a Note box about the required `storage:delete:enabled` flag when xref:../admin_guide/pruning_resources.adoc#pruning-images[pruning images].
%
|xref:../install_config/install/docker_registry.adoc#install-config-install-docker-registry[Configure or Deploy a Docker Registry]
|Added troubleshooting guidance on xref:../install_config/install/docker_registry.adoc#known-issue-prune-fails-due-to-delete-disabled[Image Pruning Failures].
%

**Pull request merged**